### PR TITLE
update documents

### PR DESCRIPTION
**Pull request by yandongxiao**
Signed-off-by: yandongxiao <yandongxiao@starrocks.com>

### CODE REVIEW

File: doc/deploy_starrocks_with_operator_howto.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    4. Using ConfigMap to configure your StarRocks cluster
+
+## Prerequisites
+
+1. Kubernetes cluster version >= 1.18.3.
+2. kubelet version >= 1.18.3.
+
+## 1. Deploy StarRocks Operator
+
+It Includes the following main steps:
```
Comment (kevincai) [inline doc/deploy_starrocks_with_operator_howto.md:23]:
```suggestion
It includes the following main steps:
```

File: doc/deploy_starrocks_with_operator_howto.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+    serviceaccount/starrocks created
+    clusterrole.rbac.authorization.k8s.io/kube-starrocks-operator created
+    clusterrolebinding.rbac.authorization.k8s.io/kube-starrocks-operator created
+    role.rbac.authorization.k8s.io/cn-leader-election-role created
+    rolebinding.rbac.authorization.k8s.io/cn-leader-election-rolebinding created
+    deployment.apps/kube-starrocks-operator created
+    ```
+2. **Deploy the StarRocks Operator by using a custom configuration file.** By default, the Operator is configured to
+   install in the starrocks namespace. To use the Operator in a custom namespace, download the Operator manifest and
+   edit all instances of namespace to specify your custom namespace.
```
Comment (kevincai) [inline doc/deploy_starrocks_with_operator_howto.md:61]:
```suggestion
   substitute all instances of namespace to your custom namespace.
```

File: doc/deploy_starrocks_with_operator_howto.md [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+    NAME                                  READY   STATUS    RESTARTS   AGE
+    starrocks-controller-65bb8679-jkbtg   1/1     Running   0          5m6s
+    ```
+
+## 2. Deploy StarRocks Cluster
+
+You need to prepare a separate yaml file to deploy the StarRocks FE, BE and CN components. You can directly use
+the [sample configuration files](https://github.com/StarRocks/starrocks-kubernetes-operator/tree/main/examples/starrocks)
+provided by StarRocks to deploy a StarRocks cluster (an object instantiated by using the custom resource StarRocks
+Cluster). For example, you can use **starrocks-fe-and-be.yaml** to deploy a StarRocks cluster that contains three FE
```
Comment (kevincai) [inline doc/deploy_starrocks_with_operator_howto.md:86]:
```suggestion
Cluster). For example, you can use **starrocks-fe-and-be.yaml** to deploy a StarRocks cluster that consists of three FE
```

File: doc/deploy_starrocks_with_operator_howto.md [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+starrockscluster-sample-be-2          1/1     Running   0          22h
+starrockscluster-sample-fe-0          1/1     Running   0          21h
+starrockscluster-sample-fe-1          1/1     Running   0          21h
+starrockscluster-sample-fe-2          1/1     Running   0          22h
+```
+
+> **Note**
+>
+> If some pods cannot start after a long period of time, you can use `kubectl logs -n starrocks <pod_name>` to view the
+> log information or use `kubectl -n starrocks describe pod <pod_name>` to view the event information to locate the
```
Comment (kevincai) [inline doc/deploy_starrocks_with_operator_howto.md:128]:
```suggestion
> log information or use `kubectl -n starrocks describe pod <pod_name>` to view the event information to address the
```

File: doc/deploy_starrocks_with_operator_howto.md [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+starrockscluster-sample-be-1          1/1     Running   0          23h
+starrockscluster-sample-be-2          1/1     Running   0          22h
+starrockscluster-sample-fe-0          1/1     Running   0          21h
+starrockscluster-sample-fe-1          1/1     Running   0          21h
+starrockscluster-sample-fe-2          1/1     Running   0          22h
+```
+
+> **Note**
+>
+> If some pods cannot start after a long period of time, you can use `kubectl logs -n starrocks <pod_name>` to view the
```
Comment (kevincai) [inline doc/deploy_starrocks_with_operator_howto.md:127]:
```suggestion
> If some pods cannot be up after a long period of time, you can use `kubectl logs -n starrocks <pod_name>` to view the
```

File: doc/deploy_starrocks_with_operator_howto.md [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+
+## 3. Manage StarRocks Cluster
+
+### 3.1. Access StarRocks Cluster
+
+The components of the StarRocks cluster can be accessed through their associated Services, such as the FE Service. For
+detailed descriptions of Services and their access addresses,
+see [api.md](https://github.com/StarRocks/starrocks-kubernetes-operator/blob/main/doc/api.md)
+and [Services](https://kubernetes.io/docs/concepts/services-networking/service/).
+
```
Comment (kevincai) [inline doc/deploy_starrocks_with_operator_howto.md:139]: add a bit more explanation on these two services. user should only use the `starrockscluster-sample-fe-service` for external usage.

File: doc/deploy_starrocks_with_operator_howto.md [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+2. Access the StarRocks cluster by using the MySQL client from within the Kubernetes cluster.
+
+   ```Bash
+   mysql -h 10.100.162.xxx -P 9030 -uroot
+   ```
+
+   Upon deploying a fresh StarRocks cluster, the `root` user's password remains unset, potentially posing a security
+   risk.
+   See [Init Root Password When First Deploy HOWTO](./initialize_root_password_howto.md) for details on how to set the
```
Comment (kevincai) [inline doc/deploy_starrocks_with_operator_howto.md:170]: here the cluster is deployed manually by operator with crd, better point to the manual change root password doc.

File: doc/deploy_starrocks_with_operator_howto.md [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+   file, and add `service` field to the `starRocksFeSpec` field.
+
+    ```YAML
+    spec:
+      starRocksFeSpec:
+        service:            
+          type: LoadBalancer # specified as LoadBalancer
+    ```
+
+2. Obtain the IP address `EXTERNAL-IP` and port `PORT(S)` that the FE Service exposes to the outside.
```
Comment (kevincai) [inline doc/deploy_starrocks_with_operator_howto.md:188]: add two more ways to access the service

- kubernetes node port
- kubectl proxy

And do a simple comparison on these 4 different ways and its suitable case.

File: doc/deploy_starrocks_with_operator_howto.md [hunk 9]
```diff
@@ -0,0 +1,297 @@
...
+Run the following command to scale out the FE cluster to 4 nodes:
+
+```bash
+kubectl -n starrocks patch starrockscluster starrockscluster-sample --type='merge' -p '{"spec":{"starRocksFeSpec":{"replicas":4}}}'
+```
+
+The scaling process lasts for a while. You can use the command `kubectl -n starrocks get pods` to view the scaling
+progress.
+
+### 3.4. Using ConfigMap to configure your StarRocks cluster
```
Comment (kevincai) [inline doc/deploy_starrocks_with_operator_howto.md:249]: Add cautions on scale-in FE nodes, 1. can only scale-in step by step, and each time less than half of the nodes can be scaled-in.

2. can't do 3->1 scale in.

File: doc/build_the_operator_images_by_yourself_howto.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+```console
+docker push ghcr.io/OWNER/starrocks-kubernetes-operator/operator:latest
+```
+
+E.g.
+Publish image to ghcr
+
+```console
+docker push ghcr.io/dengliu/starrocks-kubernetes-operator/operator:latest
```
Comment (kevincai) [inline doc/build_the_operator_images_by_yourself_howto.md:29]: update the example with our starrocks namespace as example. remove ghcr.io, use docker hub as example.

File: doc/deploy_starrocks_with_helm_howto.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
+# Deploy StarRocks With Helm
+
+We have split the kube-starrocks chart into two subcharts: operator and starrocks.
```
Comment (kevincai) [inline doc/deploy_starrocks_with_helm_howto.md:3]: indicate that splitting takes effect since v1.8.0

File: doc/mount_persistent_volume_howto.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
+# Mount Persistent Volume
+
+StarRocks Kubernetes Operator supports mounting persistent volumes to StarRocks FE and BE pods. If not specified, the
+operator will use emptyDir mode to store FE meta and BE data. **When container restarts, the data will be lost.**
+
+This document describes how to mount persistent volumes to StarRocks FE and BE pods. And There are two ways to mount
```
Comment (kevincai) [inline doc/mount_persistent_volume_howto.md:6]:
```suggestion
This document describes how to mount persistent volumes to StarRocks FE and BE pods. There are two ways to mount
```

File: doc/mount_persistent_volume_howto.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+    image: "starrocks/be-ubuntu:3.1-latest"
+    replicas: 1
+    storageVolumes:
+    - name: be-storage-data
+      storageClassName: data-storage
+      storageSize: 10Gi
+      mountPath: /opt/starrocks/be/storage
+    - name: be-storage-log
+      storageClassName: data-storage
+      storageSize: 1Gi
```
Comment (kevincai) [inline doc/mount_persistent_volume_howto.md:52]: keep this size the same as helm chart values.

File: doc/mount_persistent_volume_howto.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+      mountPath: /opt/starrocks/be/log
+```
+
+Note that the specific `storageClassName` should be available in kubernetes cluster before enabling this storageVolume
+feature. If `StorageVolume` info is not specified in CRD spec, the operator will use emptydir mode to store FE meta and
+BE data.
```
Comment (kevincai) [inline doc/mount_persistent_volume_howto.md:58]: And a FAQ or Warning notice, if the free space of fe-meta storage volume is less than 5GB, the FE service won't start.

File: doc/deploy_starrocks_with_operator_howto.md (hunk 8 above)
Comment (yandongxiao) [reply doc/deploy_starrocks_with_operator_howto.md:188]: todo

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — APPROVED** [review]

**Pull request merged**